### PR TITLE
re-organise doc

### PR DESCRIPTION
**Pull request by jburel**
Add new setup page and re-organise doc
cc @pwalczysko

### CODE REVIEW

**Comment by pwalczysko** [comment]
the README is not edited in this PR, but as it is pointed to, I found that

``This guide demonstrates how to use the CellProfiler Python API to analyze data stored in IDR or an OMERO server.`` sentence should actually say

``This guide demonstrates how to use the CellProfiler Python API to analyze data stored in IDR or on another OMERO server.``

(to make clear that IDR is also an OMERO server)

File: docs/setup.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
+Install CellProfiler
+====================
+
+In this section, we indicate how to install CellProfiler in a `Conda <https://conda.io/en/latest/>`_ environment.
```
Comment (pwalczysko) [inline docs/setup.rst]: ``indicate`` -> ``show`` (indicate might sound too light-weight)

File: docs/setup.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+
+**Setup**
+---------
+
+We recommand to install CellProfiler using Conda.
+Conda manages programming environments in a manner similar to 
+`virtualenv <https://virtualenv.pypa.io/en/stable/>`_.
+
+- We assume that you have a local copy of the `omero-guide-cellprofiler repository <https://github.com/ome/omero-guide-cellprofiler>`_. If not, first clone the repository::
```
Comment (pwalczysko) [inline docs/setup.rst]: `` We assume that you have a local copy of the `omero-guide-cellprofiler repository <https://github.com/ome/omero-guide-cellprofiler>`_. If not, first clone the repository::`` ->
``If you do not  have a local copy of the `omero-guide-cellprofiler repository <https://github.com/ome/omero-guide-cellprofiler>`_,  make it by cloning that repositoryL``
(simplify the sentence)

File: docs/setup.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    $ git clone https://github.com/ome/omero-guide-cellprofiler.git
+
+- Install `Miniconda <https://docs.conda.io/en/latest/miniconda.html>`_ if necessary.
+
+- Create a programming environment using Conda::
+
+    $ conda create -n cellprofiler python=2.7
+
+- Install CellProfiler **3.1.9** and its dependencies using an installation file::
```
Comment (pwalczysko) [inline docs/setup.rst]: But this has to be run from a specific place, I suppose 
``cd INTO THE CLONED REPO`` is missing ?

File: README.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # Guide on how to integrate CellProfiler and OMERO
 [![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/ome/omero-guide-cellprofiler/master?filepath=notebooks/idr0002.ipynb)
 
-This guide demonstrates how to use the CellProfiler Python API to analyze data stored in [IDR](https://idr.openmicroscopy.org/) or an OMERO server.
+This guide demonstrates how to use the CellProfiler Python API to analyze data stored in [IDR](https://idr.openmicroscopy.org/) or in an other OMERO server.
```
Comment (pwalczysko) [inline README.md]:
```suggestion
This guide demonstrates how to use the CellProfiler Python API to analyze data stored in [IDR](https://idr.openmicroscopy.org/) or in another OMERO server.
```

**Review by pwalczysko — APPROVED** [review]

**Comment by pwalczysko** [comment]
I thihk this PR is fine, just thinking about the doc as a whole - I would have an immediate question about how do I install https://docs.openmicroscopy.org/omero/5.6.0/developers/Python.html - in a separate conda environment or together with the CP ? We probably need a similar pointer from the Python omero-guide and somehow cross link the two....

**Comment by jburel** [comment]
I have not looked at Python guide yet.

**Comment by jburel** [comment]
You need to have both CP and omero-py in the same environment.

**Comment by pwalczysko** [comment]
Went through the workflow on my mac:

1. After the last line of the walkthrough which activates the cellprofiler venv, I tried to get the python code working. For that I did
2. Typed ``python`` -> this was wrong, because I have python3 on my machine by default, and although being in that conda env, I ended up in python3 environment -> I  think this instruction should be included here
2. After I typed ``python2``, I tried to find the pure python code of the noteboook, but that is still hidden - we need to point to https://github.com/ome/training-notebooks/blob/master/CellProfiler/idr0002.ipynb - from there, I can start to copy and paste the python code bits into my terminal. Atm, in this guide, there is only a link to the downloadable notebook, see the "notebook" bullet point in https://omero-guides.readthedocs.io/en/latest/cellprofiler/docs/cellprofiler.html - this is not enough - it would be quite tedious to get the pure python code from that

**Comment by jburel** [comment]
I will fix the creation of the env.
For point 3, I think we should look at that in another PR, it is more how to run CP from a script rather that notebook this is outside the scope of "setup". 
This is the approach I am taking for other repo instead of very large PR

**Comment by jburel** [comment]
the env should be created with python 2.7
i.e.
```conda create -n cellprofiler python=2.7```

**Comment by pwalczysko** [comment]
> the env should be created with python 2.7
i.e.
conda create -n cellprofiler python=2.7

Even as the environment is created with python 2.7, when I am in it, and type ``python``, I end up in python3 environ. That was my point. I have to type ``python2`` for that to work.

**Comment by pwalczysko** [comment]
> For point 3, I think we should look at that in another PR, it is more how to run CP from a script rather that notebook this is outside the scope of "setup".

Sorry, confused

**Comment by jburel** [comment]
Could you do ``which python`` when the env is activate and do ``python -V``?

**Comment by jburel** [comment]
cf. https://github.com/ome/omero-guide-cellprofiler/pull/6#issuecomment-589612055
I think it is better to have another PR with some scripts than having everything in the same PR

**Comment by pwalczysko** [comment]
> Could you do which python when the env is activate and do python -V?

```
(base) pwalczysko@ls31618~/Work/omero-guide-cellprofiler$ conda activate cellprofiler
(cellprofiler) pwalczysko@ls31618~/Work/omero-guide-cellprofiler$ which python
/Users/pwalczysko/miniconda3/envs/cellprofiler/bin/python
(cellprofiler) pwalczysko@ls31618~/Work/omero-guide-cellprofiler$ python -V
Python 2.7.17 :: Anaconda, Inc.
(cellprofiler) pwalczysko@ls31618~/Work/omero-guide-cellprofiler$ python
Python 2.7.17 |Anaconda, Inc.| (default, Oct 21 2019, 14:10:59) 
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> 
```

This looks actually good. I think now the previous report https://github.com/ome/omero-guide-cellprofiler/pull/6#issuecomment-589147356 was caused by me being also in a venv (outside of conda) which was python3. Now, when I am not inside this other venv, the conda environ cellprofiler is really pushing me to python2, which is as you expect.

**Pull request merged**